vdoctl/cmd/version.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cobra"
+	vdov1alpha1 "github.com/vmware-tanzu/vsphere-kubernetes-drivers-operator/api/v1alpha1"
+	"github.com/vmware-tanzu/vsphere-kubernetes-drivers-operator/controllers"
+	dynclient "github.com/vmware-tanzu/vsphere-kubernetes-drivers-operator/pkg/client"
+	vdocontext "github.com/vmware-tanzu/vsphere-kubernetes-drivers-operator/pkg/context"
+	v12 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strings"
+)
+
+const (
+	VdoDeploymentName = "vdo-controller-manager"
+)
+
+// versionCmd represents the version command
+var versionCmd = &cobra.Command{
+	Use:   "version",
+	Short: "command to get VDO version",
+	Long: `This command helps to get the version of the configurations created by VDO.
+            It includes brief detail about the version of CloudProvider and StorageProvider
+             along with VC and Kubernetes details`,
+	Run: func(cmd *cobra.Command, args []string) {
+
+		var vdoConfigList vdov1alpha1.VDOConfigList
+
+		ctx := vdocontext.VDOContext{
+			Context: context.Background(),
+			Logger:  ctrllog.Log.WithName("vdoctl:version"),
+		}
+
+		err := K8sClient.List(ctx, &vdoConfigList)
+		if err != nil {
+			cobra.CheckErr(err)
+		}
+
+		s := scheme.Scheme
+		s.AddKnownTypes(vdov1alpha1.GroupVersion, &vdov1alpha1.VDOConfig{})
+
+		r := controllers.VDOConfigReconciler{
+			Client:       K8sClient,
+			Logger:       ctrllog.Log.WithName("vdoctl:version"),
+			Scheme:       s,
+			ClientConfig: ClientConfig,
+		}
+
+		req := reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      CompatMatrixConfigMAp,
+				Namespace: VdoNamespace,
+			},
+		}
+
+		deploymentReq := reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      VdoDeploymentName,
+				Namespace: VdoNamespace,
+			},
+		}
+
+		// Fetch VDO deployment object
+		deployment := &v12.Deployment{}
+		err = K8sClient.Get(ctx, deploymentReq.NamespacedName, deployment)
+		if err != nil {
```

Comment:
would this work
	for _, container := range deployment.Spec.Template.Spec.Containers {
		container.Name == "manager" {
			container.Image
		}
	}